File: app/src/main/java/ru/org/adons/cblock/ui/adapter/BlockListAdapter.java
package ru.org.adons.cblock.ui.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import ru.org.adons.cblock.R;
import ru.org.adons.cblock.model.BlockListItem;

import static ru.org.adons.cblock.ui.adapter.ListItemDecorator.formatPhoneNumber;
import static ru.org.adons.cblock.ui.adapter.ListItemDecorator.getDescription;

/**
 * Recycler adapter for Block list
 */
public class BlockListAdapter extends RecyclerView.Adapter<BlockListAdapter.ViewHolder> {

    private final ArrayList<BlockListItem> items = new ArrayList<>();

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item, parent, false);
        return new BlockListAdapter.ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        BlockListItem item = items.get(position);
        holder.textPhone.setText(formatPhoneNumber(item.getPhoneNumber()));
        holder.textDesc.setText(getDescription(item.getName(), item.getDate()));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public void setItems(List<BlockListItem> newItems) {
        items.clear();
        items.addAll(newItems);
        notifyDataSetChanged();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.text_view_phone_number) TextView textPhone;
        @BindView(R.id.text_view_desc) TextView textDesc;

        ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

}

File: app/src/main/java/ru/org/adons/cblock/ui/view/main/BlockListAdapter.kt
package ru.org.adons.cblock.ui.view.main

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.list_item_block.view.*
import ru.org.adons.cblock.R
import ru.org.adons.cblock.data.BlockListItem
import ru.org.adons.cblock.ext.formatPhoneNumber
import ru.org.adons.cblock.ext.getDescription
import java.util.*

/**
 * Recycler adapter for Block list
 */
class BlockListAdapter(private val deleteItem: (Long) -> Unit) : RecyclerView.Adapter<BlockListAdapter.ViewHolder>() {

    private val items = ArrayList<BlockListItem>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.list_item_block, parent, false)
        return BlockListAdapter.ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]
        val view = holder.itemView
        view.textPhone.text = item.phoneNumber.formatPhoneNumber()
        view.textDesc.text = item.date.getDescription(item.name)
        view.buttonDelete.setOnClickListener { deleteItem(item.id) }
    }

    override fun getItemCount(): Int = items.size

    fun setItems(newItems: List<BlockListItem>) {
        items.clear()
        items.addAll(newItems)
        notifyDataSetChanged()
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

}
